omnigibson/objects/primitive_object.py

```diff
@@ -122,27 +122,15 @@ def _load(self):
         # Define an Xform at the specified path
         prim = og.sim.stage.DefinePrim(self._prim_path, "Xform")
 
-        if self._prim_type == PrimType.RIGID:
-            # Define a nested mesh corresponding to the root link for this prim
-            base_link = og.sim.stage.DefinePrim(f"{self._prim_path}/base_link", "Xform")
-            self._vis_geom = create_primitive_mesh(prim_path=f"{self._prim_path}/base_link/visuals", primitive_type=self._primitive_type)
-            self._col_geom = create_primitive_mesh(prim_path=f"{self._prim_path}/base_link/collisions", primitive_type=self._primitive_type)
-
-            # Add collision API to collision geom
-            UsdPhysics.CollisionAPI.Apply(self._col_geom.GetPrim())
-            UsdPhysics.MeshCollisionAPI.Apply(self._col_geom.GetPrim())
-            PhysxSchema.PhysxCollisionAPI.Apply(self._col_geom.GetPrim())
+        # Define a nested mesh corresponding to the root link for this prim
+        base_link = og.sim.stage.DefinePrim(f"{self._prim_path}/base_link", "Xform")
+        self._vis_geom = create_primitive_mesh(prim_path=f"{self._prim_path}/base_link/visuals", primitive_type=self._primitive_type)
+        self._col_geom = create_primitive_mesh(prim_path=f"{self._prim_path}/base_link/collisions", primitive_type=self._primitive_type)
 
-        elif self._prim_type == PrimType.CLOTH:
-            # For Cloth, the base link itself is a cloth mesh
-            # TODO (eric): configure u_patches and v_patches
-            self._vis_geom = create_primitive_mesh(
-                prim_path=f"{self._prim_path}/base_link",
-                primitive_type=self._primitive_type,
-                u_patches=None,
-                v_patches=None,
-            )
-            self._col_geom = None
```

Comment:
Do we not need to special case cloth anymore?

Reply:
In post_load, we dynamically delete the original link prim and replace it with the clothified mesh prim.